BeatTogether.DedicatedServer.Kernel/Managers/LobbyManager.cs

```diff
@@ -354,11 +209,134 @@ public void Update()
+                            PlayersWithoutEntitlements = _playerRegistry.Players
+                                .Where(p => p.GetEntitlement(SelectedBeatmap!.LevelId) is EntitlementStatus.NotOwned or EntitlementStatus.NotDownloaded)
+                                .Select(p => p.UserId).ToList()
+                        }, DeliveryMethod.ReliableOrdered);
+                        // Starts beatmap
+                        _gameplayManager.StartSong(SelectedBeatmap!, SelectedModifiers, CancellationToken.None);
+                        // Reset and stop counting down
+                        CountdownReset();
+                        return;
+                    }
+                }
+
+                // If manager/all players are no longer ready or not all players own beatmap
+                if (NotStartable || !allPlayersOwnBeatmap)
+                {
+                    // Reset and stop counting down
+                    CountdownReset();
+                    _packetDispatcher.SendToNearbyPlayers(new CancelCountdownPacket(), DeliveryMethod.ReliableOrdered);
+                    _packetDispatcher.SendToNearbyPlayers(new CancelLevelStartPacket(), DeliveryMethod.ReliableOrdered);
+
+                }
+                else// If manager/players is/are still ready and all players own beatmap
+                {
+                    // If all players are ready and countdown is too long
+                    if (AllPlayersReady && (CountdownEndTime - _instance.RunTime) > CountdownTimeEveryoneReady )
+                    {
+                        CountdownEndTime = _instance.RunTime + CountdownTimeEveryoneReady;
+
+                        //This caused the countdown to stop working on quest
+                        /*
+                        if (_instance.Configuration.ManagerId != "ziuMSceapEuNN7wRGQXrZg")
+                        {
+                            // Cancel countdown (bc of stupid client garbage) 
+                            _packetDispatcher.SendToNearbyPlayers(new CancelCountdownPacket(), DeliveryMethod.ReliableOrdered); //Not in old quickplay logic
+
+                            _packetDispatcher.SendToNearbyPlayers(new SetCountdownEndTimePacket                                 //Not in old quickplay logic
+                            {
+                                CountdownTime = CountdownEndTime
+                            }, DeliveryMethod.ReliableOrdered);
+                        }
+                        */
+
+                        // Set start time & start level time
+                        _packetDispatcher.SendToNearbyPlayers(new StartLevelPacket
+                        {
+                            Beatmap = SelectedBeatmap!,
+                            Modifiers = SelectedModifiers,
+                            StartTime = CountdownEndTime
+                        }, DeliveryMethod.ReliableOrdered);
+                    }
+                }
+            }
+        }
+
+        private void CountdownReset()
```

Comment:
Function that is called whenever the countdown or starting the beatmap is canceled